src/irmin/tree.ml

```diff
@@ -663,18 +663,17 @@ module Make (P : Private.S) = struct
           let+ v = to_value t in
           get_ok "length" v |> P.Node.Val.length
 
-    let is_empty t =
-      match cached_map t with
-      | Some m -> Lwt.return (Ok (StepMap.is_empty m))
-      | None -> (
-          match t.v with
-          | Value (_, v, Some um) ->
-              let res = is_empty_after_updates v um in
-              Lwt.return_ok res
-          | _ -> (
-              to_value t >|= function
```

Comment:
I'm not sure it matters but `to_value` has the advantage of testing `cached_value`